lsst-epo/rubin-obs-client
# refactor: move API endpoints to route handlers

Migrates Next endpoints from pages directory to app directory and adds tests for each handler. 

## Testing

- Preview
  - Preview an edited page from the Craft entry and verify live reload
  - Click "view" on an edited page from the Craft entry and verify the preview token is present
  - Click "view" on an unedited page from the Craft entry and verify no preview token is present
- Revalidate
  - Run a production Next build
  - Edit an entry in Craft and save
  - Refresh page in client and verify update
- Google auth
  - The fastest way to get to login/signup is to scale down the browser window until the slideout menu shows a "login/signup" option (this shouldn't be there but that's another ticket)
  - Sign up with your Noirlab email (might need to re-assign your admin account to LSST temporarily), registration should issue an activation token, if you make it to this point then the Google auth token has already worked it's magic

Run new E2E tests and validated
